### ISSUE TITLE
Clarify license

### ISSUE THREAD

**Issue by matkoniecz** [alltheplaces/alltheplaces#5133]
https://github.com/alltheplaces/alltheplaces#license

Maybe clearly mention that produced data can be still encumbered by database rights, copyright and so on?

Many people thing that CC0 means no restriction whatsoever on data use and this is sadly untrue. This applies especially in places where database rights exist and matters for example for OSM.

(if extracting for example all 4000 shop locations from website map yields data free of database rights and copyright in all cases - then it would be a good news, but as far as I know this is not true and for example alltheplaces data cannot be imported/copied into OpenStreetMap)

**Comment by matkoniecz** [comment]
And for something actionable: I propose to mention that especially database right may still apply, which may be relevant in some jurisdictions.

**Comment by iandees** [comment]
This project selected the license we did based on our interpretation of the law in our jurisdiction.

My personal policy is to not express legal opinions for other people, especially online in a recorded medium. If you don't think the project's license applies to you, then I encourage you to get legal advice from a professional (i.e. not me or this project 😄 ).

**Comment by matkoniecz** [comment]
Oh, I am not disputing that CC0 applies.

The problem is that in this case materials were crawled, so what is covered by CC0 is minimal part

> the person associating CC0 with a Work (the "Affirmer"), **to the extent that he or she is an owner of Copyright and Related Rights in the Work,** voluntarily elects to apply CC0 to the Work

and current formulation suggests that this data can be freely reused, which - as far as I know - is not true.

Especially given

> Many contributors are also present on [OSM US Slack](https://slack.openstreetmap.us/), in particular we watch the [#poi](https://osmus.slack.com/archives/CDJ4LKA2Y) channel.

I hope that noone is importing that into OpenStreetMap database - directly or indirectly.

**Comment by matkoniecz** [comment]
> My personal policy is to not express legal opinions for other people, especially online in a recorded medium. 

Maybe quoting relevant license parts in that section would still be fine?

Note that CC0 applies "to the extent that he or she is an owner of Copyright and Related Rights in the Work" and "A Work made available under CC0 may be protected by copyright and related or neighboring rights ("Copyright and Related Rights").".

**Comment by Cj-Malone** [comment]
IIRC years ago I contacted you about some PL data, that was before I knew about ATP, but ATP is essentially the same as that.

> I hope that noone is importing that into OpenStreetMap database - directly or indirectly.

Someone would have made a big fuss about it by now. We (OSM Community) can't even keep OGL data in sync, or standardise surface values without a ton of drama :p 2M POIs definitely can't imported without anyone noticing.

FYI in the outputted data and the code we document [explicit licences](https://github.com/alltheplaces/alltheplaces/blob/8881e703e547db348b846dbe0fbd553ad3c315d7/locations/spiders/bern_ch.py#L19), as well as implicitly [published](https://github.com/alltheplaces/alltheplaces/blob/8881e703e547db348b846dbe0fbd553ad3c315d7/locations/structured_data_spider.py#L53) data. Data consumers can make up there own mind on what data to use.

I defiantly think this data should be used in the OSM world. Small scale in GB, we use it for [QA](https://osm.mathmos.net/chains/). And I'd love for this data to be more used by OSM consumers, I want it to be in OsmAnd. I want it to be a sidecar file for Daylight. I want this data to improve my life.

**Comment by mnalis** [comment]
> Note that CC0 applies "to the extent that he or she is an owner of Copyright and Related Rights in the Work" and "A Work made available under CC0 may be protected by copyright and related or neighboring rights ("Copyright and Related Rights").".

That is true. The CC0 applies only to the rights of the _Affirmer_, i.e.:

> Affirmer disclaims responsibility for clearing rights of other persons that may apply to the Work or any use thereof, including without limitation any person's Copyright and Related Rights in the Work. Further, Affirmer disclaims responsibility for obtaining any necessary consents, permissions or other rights required for any use of the Work.

However, if the _Affirmer_ is distributing the data illegally (i.e. the distributing the data they do not have rights to distribute), then the Affirmers license is irrelevant, as what is important is the lack of license (or other allowed rights) of original work that the Affirmer derived their database from.

>  (if extracting for example all 4000 shop locations from website map yields data free of database rights and copyright in all cases - then it would be a good news, but as far as I know this is not true and for example alltheplaces data cannot be imported/copied into OpenStreetMap)

IANAL, but it is my understanding that scraping such miniscule amount of data from websites is not covered by copyright or database rights or other related right themselves in any country (fair use, unprotectability of factual data and other similar exceptions in different laws). After all, if for example `opening_hours` data was protected by copyright or related rights, it would be irrelevant whether you copied that data from the sign on the door, or by looking up the website and copying `opening_hours` data from there - it would be illegal to collect and enter that data in OSM in both cases.

So it is my interpretation that such scraped data should be free of the restrictions. But the person/entity who scraped and created that database, might have created new rights for themselves depending on jurisdiction (database rights in EU, sweat of the brow doctrine etc.) -- it is **that rights** that CC0 by alltheplaces waves away. And if those are the only rights to contend with, that using them should be OK.

(it would be completely another story if alltheplaces were scraping one homogeneous already-compiled database like e.g. phonebook/yellow pages; then their database would be derived work of that pre-existing database; and copyright/database rights might apply depending on the jurisdiction).

> And, yes, I am not a lawyer - but if I would consult lawyer before doing anything copyright related I would go bankrupt within hours.

That being said, whether such mass processing of data is allowed, is still somewhat controversial - but since GitHub copilot still exists and Microsoft hasn't been shut down yet (and presumably their legal has thrown some serious resources into researching it), it seems it should be OK:  https://en.wikipedia.org/wiki/GitHub_Copilot#Licensing_controversy

---

TL;DR: I am not a lawyer, but it seems to be that using that data in OSM should be OK. (But as always, for anything you do, if you want to be absolutely sure, you need to consult relevant lawyer to clear you, and not depend on hearsay from random people on the Internet.)

**Comment by matkoniecz** [comment]
> After all, if for example opening_hours data was protected by copyright or related rights, it would be irrelevant whether you copied that data from the sign on the door, or by looking up the website and copying opening_hours data from there

I suspect that there may be distinction between opening_hours on a door and opening hours dataset displayed at the website, with second qualifying for protection - but I am not sure about it, hope that it does not apply and I am not a lawyer

But maybe listing such as https://www.biedronka.pl/pl/sklepy/lista,city,warszawa already qualifies as a protected database?

**Comment by Cj-Malone** [comment]
> But maybe listing such as https://www.biedronka.pl/pl/sklepy/lista,city,warszawa already qualifies as a protected database?

I believe Biedronka will be more annoyed if OSM has wrong hours than if we copied them off the door or the website. Loss in reputation if someone turns up but it's actually closed, loss in money if someone doesn't go because OSM said it's closed.

**Comment by matkoniecz** [comment]
The question is not whether it would be helpful/useful to have that data, the question is whether it is 100% legal to do it without permission.

**Comment by Cj-Malone** [comment]
> if for example `opening_hours` data was protected by copyright or related rights, it would be irrelevant whether you copied that data from the sign on the door, or by looking up the website and copying `opening_hours` data from there - it would be illegal to collect and enter that data in OSM in both cases.

This was my understanding, but maybe if someone feels strongly, instead of us internet lawyering they could convenience OSMF to commission a real lawyer.

**Comment by hfs** [comment]
@matkoniecz Maybe you could take this question to the `legal-talk` list to get some kind of “official consensus” on the matter that can be referred to?

**Comment by matkoniecz** [comment]
That would be probably a good idea, though I expect it to be done by people who want to import this data into OSM (as confirming that data is on license allowing import into OSM is a prerequisite for valid import).

**Comment by davidhicks** [comment]
Important to note is that AllThePlaces is not just a copy of source data in its original state, order and presentation. Much of the source data is riddled with obvious errors that AllThePlaces spiders generally correct. And AllThePlaces changes the presentation of data to be consistent and machine readable. Examples are converting phone numbers to ITU E.164 notation, changing the case of store names and removing superfluous words like "NOW OPEN", reformatting open hours into an OSM-like notation, etc. AllThePlaces ignores a lot of source data that is deemed irrelevant, such as the model of a point of sale system used at a shop. There is no attempt made to preserve the order or presentation of source data.

If ATP data were to be imported into OSM, further transformation would usually be necessary. Coordinates provided in source data are generally imprecise (+/-100m). An OSM editor may be able to use other data such as government-provided postal address data or satellite imagery to improve precision to +/-1m. An OSM editor may trace an entire building as an area and use this area to apply ATP-sourced tags, effectively discarding the ATP coordinates. OSM tools may then adjust address data provided from ATP to be consistent with government-provided postal address data, or consistent with other map features. For example, if "Somethng St" (missing "i") appears in ATP data, by the time it reaches OSM, this would likely to have been fixed to "Something St" to match all the other nearby locations on the same road feature.

In summary, the data that ends up in OSM would look completely different to the source data, generally due to data having been corrected and correlated with other data sources to improve accuracy.

**Comment by matkoniecz** [comment]
> In summary, the data that ends up in OSM would look completely different to the source data

If initial work is protected by some copyright-like protection, then it remains a derivative work of that initial data.

Additional AllThePlaces processing could add more restrictions which are waived by CC0 but it cannot remove source data restrictions in case they exist.

**Comment by Cj-Malone** [comment]
If anyone knows of any specific copyright/database rights relating to an ATP spider, please let me know. I'll open a PR to document that and pass it along to out consumers.

> (if extracting for example all 4000 shop locations from website map yields data free of database rights and copyright in all cases - then it would be a good news, but as far as I know this is not true)

OSM LWG disagrees.

**Comment by mnalis** [comment]
for reference: https://mastodon.online/@CjMalone@en.osm.town/111030341889346005, but please update it with the actual reference when LWG does publish the decision (_"in few months"_).

**Comment by riQQ** [comment]
See also https://osmfoundation.org/wiki/Licensing_Working_Group/Minutes/2023-08-14#Ticket#2023081110000064_%E2%80%94_First_party_websites_as_sources
